temporal-sdk/src/main/java/io/temporal/internal/sync/SyncWorkflowContext.java

```diff
@@ -354,7 +354,10 @@ private ExecuteActivityParameters constructExecuteActivityParameters(
                 ProtobufTimeUtils.toProtoDuration(options.getStartToCloseTimeout()))
             .setScheduleToCloseTimeout(
                 ProtobufTimeUtils.toProtoDuration(options.getScheduleToCloseTimeout()))
-            .setHeartbeatTimeout(ProtobufTimeUtils.toProtoDuration(options.getHeartbeatTimeout()));
+            .setHeartbeatTimeout(ProtobufTimeUtils.toProtoDuration(options.getHeartbeatTimeout()))
+            .setRequestEagerExecution(
+                !options.isEagerExecutionDisabled()
+                    && Objects.equals(taskQueue, replayContext.getTaskQueue()));
```

Comment:
What is the purpose of this equals check?

Reply:
Do not request eager activity execution if the activity's task queue is not the same as the workflow's activity queue. This check is actually redundant (it is done again in `tryReserveActivitySlot`). Doing it at this point is just an optimization, as it avoids the probability that we will need to "alter" the Schedule Activity command later on.

Reply:
Gotcha, I was confused by the term "replayContext" but I see in the code above, this is what the unset task queue defaults to, :+1: